portafly/src/components/pages/accounts/listing/AccountsListingTable.tsx

```diff
@@ -0,0 +1,138 @@
+  }
+
+  const { startIdx, endIdx, resetPagination } = useDataListPagination()
+  const { filters } = useDataListFilters()
+  const { modal } = useDataListBulkActions()
+
+  const options = [
+    { name: 'approved', humanName: t('actions_filter_options.by_state_options.approved') },
+    { name: 'pending', humanName: t('actions_filter_options.by_state_options.pending') },
+    { name: 'rejected', humanName: t('actions_filter_options.by_state_options.rejected') },
+    { name: 'suspended', humanName: t('actions_filter_options.by_state_options.suspended') }
+  ]
+
+  const categories = [
+    { name: 'group', humanName: t('accounts_table.group_header') },
+    { name: 'admin', humanName: t('accounts_table.admin_header') },
+    { name: 'state', humanName: t('accounts_table.state_header'), options }
+  ]
+
+  useEffect(() => resetPagination, [filters])
+
+  const actions = {
+    sendEmail: t('shared:bulk_actions.send_email'),
+    changeState: t('shared:bulk_actions.change_state')
+  }
+
+  const filteredRows = filterRows(rows, filters, columns)
+
+  const visibleRows = filteredRows.slice(startIdx, endIdx)
+
+  const onSort: OnSort = (_event, index, direction) => {
+    setSortBy(index, direction, true)
+  }
+
+  const onSelectOne: OnSelect = (_ev, selected, _rowIndex, rowData) => {
+    selectOne(rowData.id, selected)
+  }
+
+  const pagination = <PaginationWidget itemCount={filteredRows.length} />
+
+  const Header = (
+    <Toolbar>
+      <ToolbarContent>
+        <ToolbarItem>
+          <BulkSelectorWidget />
+        </ToolbarItem>
+        <ToolbarItem>
+          <BulkActionsWidget actions={actions} />
+        </ToolbarItem>
+        <ToolbarItem>
+          <SearchWidget categories={categories} />
+        </ToolbarItem>
+        <ToolbarItem variant="pagination" breakpointMods={[{ modifier: 'align-right', breakpoint: 'md' }]}>
+          {pagination}
+        </ToolbarItem>
+      </ToolbarContent>
+    </Toolbar>
+  )
+
+  return (
+    <>
+      <Table
+        aria-label={t('accounts_table.aria_label')}
+        header={Header}
+        cells={columns}
+        rows={visibleRows}
+        onSelect={visibleRows.length ? onSelectOne : undefined}
+        canSelectAll={false}
+        sortBy={sortBy}
+        onSort={onSort}
+      >
+        <TableHeader />
+        <TableBody />
+      </Table>
+      <Toolbar>
+        <ToolbarContent>
+          <ToolbarItem variant="pagination" breakpointMods={[{ modifier: 'align-right', breakpoint: 'md' }]}>
+            {pagination}
+          </ToolbarItem>
+        </ToolbarContent>
+      </Toolbar>
+
+      {modal === 'sendEmail' && (
+        <SendEmailModal
+          items={selectedRows.map((r) => `${(r.cells as string[])[1]} (${(r.cells as string[])[0]})`)}
+        />
+      )}
+
+      {modal === 'changeState' && (
+        <ChangeStateModal
+          items={selectedRows.map((r) => `${(r.cells as string[])[0]} (${(r.cells as string[])[4]})`)}
```

Comment:
I find it a bit cryptic the indexes of the arrays `0` and `4`

Reply:
It is super cryptic... I've been wanting to do something like:
```jsx
`${(r.cells.group} (${r.cells.admin})`
```
I guess we could give those indexes names but.. I would leave it for the next iteration (it would involve changes in the sorting and table data factory at least).

Reply:
We could wrap the cells into a AccountRow entity, for the sake of readability, but we would just be moving the "crypticity" somewhere else.
```jsx
const { group, admin } = new Row<IDeveloperAccount>(r.cells)
```

Reply:
Hide it behind a fn with a name that tell us the story is good enough